### DISCUSSION TITLE
False positive error? on DocumentSnapshot.data() about inline code with incorrect jvm version

### DISCUSSION THREAD

**Discussion by AlbRoehm** [GitLiveApp/firebase-kotlin-sdk#464] [Q&A] [answered]
I use the following code to fetch the UserData from a firebase collection. It works fine for JVM-Desktop (i did not come around to integrate it on Android), despite AndroidStudio is giving the error below. It's unhappy about `it.data<UserData>()`. Should i rather use `it.data(UserData.serializer())` ?

```kotlin
class UserDataService(
    private val firestore: FirebaseFirestore,
) : IUserDataService {
    override suspend fun getUserData(userId: String): Flow<UserData> = queryPath(userId).snapshots.map { it.data<UserData>() }
    private fun queryPath(userId: String) = firestore.collection(USER_DATA_COLLECTION).document(userId)
}
```

Cannot inline bytecode built with JVM target 17 into bytecode that is being built with JVM target 1.8. Please specify proper '-jvm-target' option

Any feedback appreciated.

**Comment by nbransby** [comment, marked as answer]
Not sure, is it this? https://youtrack.jetbrains.com/issue/KTIJ-25134

**Reply by AlbRoehm** [reply]
Probably, seems quite to be a quite similar thing. 
I found that using `queryPath(userId).snapshots.map { it.data(UserData.serializer()) }` works as well but does not raise any errors or warnings. Seems to be fine as well i guess.

**Comment by tellypresence** [comment]
Ran into the same problem; in my case was able to solve as indicated in issue #736
